lib/rouge/lexers/lustre.rb

```diff
@@ -0,0 +1,90 @@
+# -*- coding: utf-8 -*- #
+#
+# adapted from ocaml.rb, hence some ocaml-ism migth remains
+module Rouge
+  module Lexers
+    class Lustre < RegexLexer
+      title "Lustre"
+      desc 'The Lustre programming language (Verimag)'
+      tag 'lustre'
+      filenames '*.lus'
+      mimetypes 'text/x-lustre'
+
+      def self.keywords
+        @keywords ||= Set.new %w(
+          extern unsafe assert const current enum function
+          false let node operator returns
+          step struct tel type true var  model package needs
+          provides uses is body end include merge
+        )
+      end
+
+      def self.word_operators
+        @word_operators ||= Set.new %w(
+           div and xor mod or not nor if then else fby pre when with) 
+      end
+
+      def self.primitives
+        @primitives ||= Set.new %w(int real bool)
+      end
+
+      operator = %r([;,_!$%&*+./:<=>?@^|~#-]+)
+      id = /[a-z_][\w']*/i
+
+      state :root do
+        rule /\s+/m, Text
+        rule /false|true|[(][)]|\[\]/, Name::Builtin::Pseudo
```

Comment:
This doesn't look like the right tokens to use for these values. `true` and `false` should be `Keyword::Constant`. As for `()` and `[]`, these are normally just left to be highlighted as punctuation (you have a rule later that will do that).

Reply:
Ok.
Basically, I've taken inspiration from ocaml.rb, since ocaml is a langage I know.